lib/circuit.js

```diff
@@ -663,7 +663,7 @@ function fallback (circuit, err, args) {
 }
 
 function fail (circuit, err, args, latency) {
-  if (circuit.options.errorFilter(err)) return;
+  if (!circuit.halfOpen && circuit.options.errorFilter(err)) return;
```

Comment:
I meant to include this in my review. This seems wrong because, for example, in your test, the filter will ignore anything with an error code < 500 - except in the case where the circuit is in the half-open state. I think it would be better to reset `PENDING_CLOSE`.
```suggestion
  if (!circuit.halfOpen && circuit.options.errorFilter(err)) {
    if (circuit.halfOpen) circuit[PENDING_CLOSE] = true;
    return;
  }
```

Reply:
That was my first implementation, but it seemed good neither because the circuit could accept a lot of requests while in the half-open state. But thinking again it seems a better solution indeed.

Another option would be going straight to the closed state, as a filtered error usually is an expected result. What do you think?

Reply:
> Another option would be going straight to the closed state, as a filtered error usually is an expected result. What do you think?

Actually, this is a good point. If there is an error filter in place, when an error is filtered it is considered a success condition. The circuit should probably be opened at that point.

Reply:
If the error should be considered a success, shouldn't we emit `success` instead of `failure`?

```suggestion
  if (circuit.options.errorFilter(err)) {
    circuit.emit('success', err, latency);
    return;
  }
```
But err as result doesn't seem right, does it?

Reply:
I have to think about this. You are right, `success` should probably be emitted - though that has never been the case with filtered errors in the past. Not sure whether passing the `Error` object makes sense.

Reply:
I think the circuit should emit `success` and provide the error. It's a little weird, but providing the error at least gives some information.